examples/basic_workflow.py

```diff
@@ -0,0 +1,107 @@
+"""End‑to‑end augmentation script executed from the repo root.
+
+Usage
+-----
+python examples/augment_pipeline.py \
+       --dataset_root datasets/ \
+       --exercise deep_squat \
+       --out augmented/ \
+       --beta_delta angle_series/beta_delta.npz \
+       --ik_model models/rajagopal_scaled.osim
+"""
+from __future__ import annotations
+
+from src.imu_augment_sim.dataset import ExerciseDataset
+from src.imu_augment_sim.preprocessing import calibrate_segments, calculate_distribution_parameters_from_data, convert_to_imu
+from src.imu_augment_sim.augmentation import augment_sample
+from src.imu_augment_sim.ik import create_opensim_file, perform_ik
+from src.imu_augment_sim.labeling import RuleEvaluator
+from src.imu_augment_sim.analysis import load_analysis_data, perform_analysis
+from pathlib import Path
+import json
+from util import load_example_data
+import yaml
+from example_al_rules import al_rules, al_logic
+
+
+# -----------------------------------------------------------------------------
+# MAIN
+# -----------------------------------------------------------------------------
+
+def main():
+    """Run the full augmentation and analysis example.
+
+    This script demonstrates the end-to-end pipeline from data loading to
+    automatic labelling. It is intended to be executed from the repository
+    root.
+
+    Returns
+    -------
+    None
+        The function is executed for its side effects.
+    """
+    with open('example_config.yml', 'r') as stream:
+        cfg = yaml.safe_load(stream)
```

Comment:
```suggestion
    with open('example_config.yml', 'r') as example_conf:
        cfg = yaml.safe_load(example_conf)
```
a stream is (at least for me) something different.

Furthermore, I'd move those two lines down to where cfg is first used (above line 54). This way file reading is done when needed. Currently two major operations are in between loading the config and using it. They might fail, which makes this operation useless.